From 9e3cec5c7e9a2f4a435039e45b73ab50f52bb681 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 16 Sep 2026 01:39:33 +0800
Subject: [PATCH] init_chainablestack.hpp: use sys/qos.h conditionally

Fixes regression from https://github.com/stan-dev/math/commit/abf876b14aa3b9c2cc186ebdf139523a4880b881
<sys/qos.h> may not exist, and it is only needed for aarch64, so include it conditionally.
---
 stan/math/rev/core/init_chainablestack.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/stan/math/rev/core/init_chainablestack.hpp b/stan/math/rev/core/init_chainablestack.hpp
index 511cef41990..cbe5bba8e03 100644
--- stan/math/rev/core/init_chainablestack.hpp
+++ stan/math/rev/core/init_chainablestack.hpp
@@ -13,8 +13,10 @@
 
 #ifdef __APPLE__
 #include <pthread.h>
+#if defined(__arm64__) || defined(__aarch64__)
 #include <sys/qos.h>
 #endif
+#endif
 
 namespace stan {
 namespace math {
